src/Pages/CreateProduct/components/ProductForm/index.tsx

```diff
@@ -21,19 +27,42 @@ const Root = styled.div`
 const createProduct = (product: ICreateProduct) =>
   ProductService.createProduct(product);
 
-export const ProductForm = () => {
+export const ProductForm = ({
+  productData,
+  handleSuccess,
+  onCancel,
+  onDone,
+}: IProps) => {
   const { t } = useTranslation();
-  const navigation = useNavigate();
-  const { mutate, isLoading } = useMutation(createProduct, {
-    onSuccess: () => navigation(URL.PRODUCTS),
+  const editProduct = (product: ICreateProduct) =>
+    ProductService.editProduct(productData?.id || "", product);
+
+  const { mutate: createMutate, isLoading: isLoading } = useMutation(
+    createProduct,
+    {
+      onSuccess: handleSuccess,
+    }
+  );
+
+  const { mutate: editMutate } = useMutation(editProduct, {
+    onSuccess: () => {
+      handleSuccess;
+      onDone;
+    },
   });
```

Comment:
Please send `onSubmit` from outside and remove all those.

```suggestion
```

Reply:
Fixed